daemon/cluster/executor/container/controller.go

```diff
@@ -155,7 +155,11 @@ func (r *controller) Prepare(ctx context.Context) error {
 			return exec.ErrTaskPrepared
 		}
 
-		return err
+		if r.pullErr != nil {
```

Comment:
Wondering if `err.Error()` in this case ever returns something different than `No such image:  foo:bar`. If not, perhaps we should use;

```go
return errors.Wrap(r.pullErr, "pulling image failed")
```

Reply:
@thaJeztah Thank you for your feedback.

In a specific situation, `err.Error()` could return another message than `No such image: foo:bar`.
For example, when
- pulling an image is failed (cause unauthorized),
- and the image exists on worker locally,
- and verifying container settings is failed.

```bash
$ docker service create --name t1 -w invalid/path registry.navercorp.com/albamc/stress:latest
image registry.navercorp.com/albamc/stress:latest could not be accessed on a registry to record
its digest. Each node will access registry.navercorp.com/albamc/stress:latest independently,
possibly leading to different nodes running different
versions of the image.

$ docker service ps --no-trunc t1
ID                          NAME                IMAGE                                         NODE                  DESIRED STATE       CURRENT STATE                     ERROR                                                                                PORTS
na2a793vmt8ayz43jjo930jnl   t1.1                registry.navercorp.com/albamc/stress:latest   dev-dcluster002.ncl   Ready               Rejected less than a second ago   "the working directory 'invalid/path' is invalid, it needs to be an absolute path"
```

With this PR, the ERROR field would be like below.
```bash
$ docker service ps --no-trunc t1
ID                          NAME                IMAGE                                         NODE                  DESIRED STATE       CURRENT STATE                     ERROR                                                                                                                                                PORTS
m5849dimorvcark3yzb5pw5q5   t1.1                registry.navercorp.com/albamc/stress:latest   dev-dcluster001.ncl   Ready               Assigned less than a second ago
bk85qyngd00sg53zn16pk8iim    \_ t1.1            registry.navercorp.com/albamc/stress:latest   dev-dcluster001.ncl   Shutdown            Rejected 4 seconds ago            "unauthorized: The client does not have permission for manifest: the working directory 'invalid/path' is invalid, it needs to be an absolute path"
```
I tested it with docker version 17.10.0-ce (commit f4ffd25).